src/Intervention/Image/Gd/Commands/ResetCommand.php

```diff
@@ -17,7 +17,7 @@ public function execute($image)
     {
         $backupName = $this->argument(0)->value();
 
-        if (is_resource($backup = $image->getBackup($backupName))) {
+        if (is_resource($backup = $image->getBackup($backupName)) || (is_object($backupName) && $backupName instanceof \GDImage)) {
```

Comment:
It would be cleaner to move the `$backup` var assignment before the `if` statement.

Reply:
According to https://php.watch/versions/8.0/gdimage it's not necessary to call is_object.
Also you should use `$image->getBackup($backupName) instanceof \GDImage`, otherwise the default reset and backup commands are not working.